app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/continuousdelivery/ContinuousDeliveryHandler.java

```diff
@@ -0,0 +1,184 @@
+import javax.ws.rs.core.StreamingOutput;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiParam;
+import io.syndesis.common.model.connection.ConnectionOverview;
+import io.syndesis.common.model.integration.ContinuousDeliveryEnvironment;
+import io.syndesis.common.model.integration.ContinuousDeliveryImportResults;
+import io.syndesis.common.model.integration.IntegrationDeployment;
+import io.syndesis.common.model.monitoring.IntegrationDeploymentStateDetails;
+
+@Api("continuous-delivery")
+@Path("/public/continuousdelivery")
+public interface ContinuousDeliveryHandler {
+
+    /**
+     * List all available environments.
+     */
+    @GET
+    @Path("environments")
+    @Produces(MediaType.APPLICATION_JSON)
+    List<String> getReleaseEnvironments();
+
+    /**
+     * List all tags associated with this integration.
+     */
+    @GET
+    @Path("releasetags/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    Map<String, ContinuousDeliveryEnvironment> getReleaseTags(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId);
+
+    /**
+     * Delete an environment tag associated with this integration.
+     */
+    @DELETE
+    @Path("releasetags/{id}/{env}")
+    void deleteReleaseTag(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId, @NotNull @PathParam("env") @ApiParam(required = true) String environment);
+
+    /**
+     * Tag an integration for release to target environments.
+     */
+    @POST
+    @Path("releasetags/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    Map<String, ContinuousDeliveryEnvironment> tagForRelease(@NotNull @PathParam("id") @ApiParam(required = true) String integrationId,
+                                                @NotNull @ApiParam(required = true) List<String> environments);
+
+    /**
+     * Export integrations to a target environment.
+     */
+    @GET
+    @Path("export/{env}/export.zip")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    StreamingOutput exportResources(@NotNull @PathParam("environment") @ApiParam(required = true) String environment,
+                           @QueryParam("all") @ApiParam() boolean exportAll) throws IOException;
+
+    /**
+     * Import integrations into a target environment.
+     */
+    @POST
+    @Path("import")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    ContinuousDeliveryImportResults importResources(@Context SecurityContext sec, @NotNull @ApiParam ImportFormDataInput formInput) throws IOException;
+
+    /**
+     * Configure a connection.
+     */
+    @POST
+    @Path("connections/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    ConnectionOverview configureConnection(@Context SecurityContext sec, @NotNull @PathParam("id") @ApiParam String connectionId, @NotNull @ApiParam Map<String, String> properties) throws IOException;
```

Comment:
Is this needed?

Reply:
@chirino suggested that the CD pipeline might want to do import/export separate from configuring connections, e.g. calling this endpoint for each connection to be configured.

Reply:
I'd split this into three feature sets, one for import/export (addressed here), one for managing connection configuration and one for managing integration state. Code-wise that would be three handlers I think...

Reply:
We could do it that way. But I was concerned that we would end up with many handlers with one or two shallow methods. Do we want to have to give out lots of simple Open API docs, or a single public API doc that captures everything?

Reply:
We have single (well, two) OpenAPI docs in the runtime right now for a lot of handlers, so that should not be an issue, checkout:

https://github.com/syndesisio/syndesis/blob/1f349f4b87f389b4cb50428002b03d3255082be2/app/server/runtime/pom.xml#L256-L297

Reply:
Added package `io.syndesis.server.endpoint.v1.handler.external` to swagger docs. 
The public API handler can be refactored into multiple handlers in the future if the number of APIs increases significantly.